### PR TITLE
Update getRights and getLogo for v3 manifests

### PR DESCRIPTION
**Pull request by jamesmisson**
Adding a getRights method, and fixing the getLogo method to get metadata from v3 manifests. This will allow subsequent changes to manifold and iiif-metadata-component that will address the issue of v3 manifest metadata not appearing the the UV metadata panel: https://github.com/UniversalViewer/universalviewer/issues/973

### CODE REVIEW

File: src/IIIFResource.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export class IIIFResource extends ManifestResource {
         return null;
       }
 
-      logo = provider.logo;
+      logo = provider[0].logo;
```
Comment (stephenwf) [inline src/IIIFResource.ts:90]: I think it could be either

Comment (demiankatz) [reply src/IIIFResource.ts:90]: Agreed, it's most backward-compatible to account for both possibilities.

Comment (jamesmisson) [reply src/IIIFResource.ts:90]: The v3 specs say that the provider property must be an array of objects, so I don't think provider.logo would work. V2 manifests are still accounted for in lines 77, 93-99.

https://iiif.io/api/presentation/3.0/#metadata:~:text=the%20seeAlso%20section.-,The,-value%20must%20be

Comment (demiankatz) [reply src/IIIFResource.ts:90]: Ahh, I see -- thanks for the extra context. Should we change `provider[0].logo` to `provider[0]?.logo` since logo is an optional element? Should we perhaps search the provider array until we find the first logo?

Comment (jamesmisson) [reply src/IIIFResource.ts]: Thanks Demian. I've adjusted this line to give the first item in the array with a logo, if it exists. If none exists then the code below handles it.

Comment (demiankatz) [reply src/IIIFResource.ts]: Isn't `provider.find` going to return a whole `provider` object, so that you need to index into it to get the `logo` part? Or am I misinterpreting something?

I wonder if this is something we could add unit test coverage for to be sure the edge cases work as expected, too. I think testing is fairly straightforward in this project, though I confess I haven't looked in years. Happy to refresh my memory and help as needed.

Comment (jamesmisson) [reply src/IIIFResource.ts]: `provider` is an array of objects, and `provider.find` will return the first object in the array with a logo property, so I think that's what we want.

Yes would be great to think about unit testing for this. In general I've found it extremely difficult to test the manifesto>manifold>iiif-metadata-pipeline as the environment was very hard to set up and is temperamental. In hindsight the original UV issue should have been split into distinct issues for each module.

Comment (demiankatz) [reply src/IIIFResource.ts]: @jamesmisson, my point is that if `provider` is something like `[{logo: "foo}]` then the `provider.find` call should return `{logo: "foo"}`, but I think the subsequent code is expecting just `"foo"`. I think there may need to be an extra line like `logo = logo.logo ?? logo;` or equivalent to unpack the object.

Regarding testing, it looks like there are Mocha tests in the [tests directory](https://github.com/IIIF-Commons/manifesto/tree/master/test). I think you can add fixtures to the fixtures folder and then add tests to the tests folder to act on those fixtures -- it should be possible to build a few samples with different permutations of logos and then confirm that they work as expected. I think `npm run test` should run the suite. I'm personally much more familiar with Jest than Mocha, but the test structure looks nearly identical.

Comment (jamesmisson) [reply src/IIIFResource.ts]: Thanks @demiankatz , yes this line is intended to return `{logo: "foo"}`, and this is unpacked by the subsequent code. Unless I'm misunderstanding, line 99 does what you're suggesting, i.e. unpacking the id property (a URL string) from the logo and returning it: `return logo["@id"] || logo.id;` . If `logo` is already a string before this point (i.e. in the case of v2 manifests) then it's returned in line 95.

Comment (demiankatz) [reply src/IIIFResource.ts]: The unpacking doesn't seem to account for the possibility that the desired value will be in a `logo` property of the `logo` variable. I _think_ that might be an issue, but maybe I misunderstand the structure of the data. I'd definitely encourage taking the time to add a test, since that should settle one way or the other whether further adjustments are needed. Let me know if you need any help with that!

Comment (jamesmisson) [reply src/IIIFResource.ts]: Aha I understand now — that was my misunderstanding of the structure! I've amended the line and unpacked it to the relevant level before continuing with the code. I've added "agent" so the code better reflects the structure in the specs. Next week I'll get the tests set up and check this (it currently works in my local version of manifesto with manual testing).

File: src/IIIFResource.ts [hunk 2]
```diff
@@ -87,7 +87,9 @@ export class IIIFResource extends ManifestResource {
         return null;
       }
 
-      logo = provider.logo;
+      // get the first agent in the provider array with a logo
+      const agent = provider.find(item => item.logo !== undefined);
+      logo = agent.logo
```
Comment (demiankatz) [inline src/IIIFResource.ts:92]: One last suggestion: you might want to make this `agent?.logo` to be null-safe, in case `provider.find` doesn't find anything.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @jamesmisson -- one last suggestion and then I think this is in good shape... though I see that I don't have merge rights here, so we'll have to figure out how to move it forward.

**Review by demiankatz — APPROVED** [review]
